api/client.go

```diff
@@ -71,7 +74,12 @@ func (c *Client) Do(req *http.Request, v interface{}) (*http.Response, error) {
 	default:
 		if v != nil {
 			defer res.Body.Close()
-			if err := json.NewDecoder(res.Body).Decode(v); err != nil {
+
+			var bodyCopy bytes.Buffer
+			body := io.TeeReader(res.Body, &bodyCopy)
+
+			if err := json.NewDecoder(body).Decode(v); err != nil {
+				debug.Println("res.Body:", bodyCopy.String())
 				return nil, fmt.Errorf("error parsing API response - %s", err)
```

Comment:
I over-condensed the above. This is the real command:

    go build ./exercism/main.go && ./main --verbose -c config.json fetch go leapa

Cool. I'll space out the responses.